BeeSwift/CurrentUserManager.swift

```diff
@@ -173,6 +173,39 @@ class CurrentUserManager : NSObject {
             error?(responseError)
         }
     }
+    
+    
+    /// fetches a goal
+    /// - Parameters:
+    ///   - slug: slug corresponding to goal to be fetched
+    ///   - success: callback containing the fetched goal
+    ///   - error: callback containing the error
+    func fetchGoal(_ slug: String, success: ((_ goal : JSONGoal) -> ())? = nil, error: ((_ error : Error? ) -> ())? = nil) {
+        guard let username = self.username else { return }
+    
+        RequestManager.get(url: "/api/v1/users/\(username)/goals/\(slug)?datapoints_count=5", parameters: nil, success: { (responseObject) in
+            
+            guard let responseObject = responseObject else { return }
+            let json = JSON(responseObject)
+            let jsonGoal = JSONGoal(json: json)
+            
+            let index = self.goals.firstIndex { jGoal -> Bool in
+                jGoal.slug == jsonGoal.slug && jGoal.id == jsonGoal.id
```

Comment:
this might not be consistent with code elsewhere

true the slug is to be unique among the user's list of slugs
but goals (slugs) can be renamed

the id is being checked here which should survive renames but goes beyond the mere slug comparison elsewhere